src/backend/base/langflow/base/tools/component_tool.py

```diff
@@ -78,19 +174,56 @@ def get_tools(self) -> list[BaseTool]:
 
             output_method: Callable = getattr(self.component, output.method)
             args_schema = None
+            tool_mode_inputs = [_input for _input in self.component.inputs if getattr(_input, "tool_mode", False)]
             if output.required_inputs:
                 inputs = [self.component._inputs[input_name] for input_name in output.required_inputs]
+                # If any of the required inputs are not in tool mode, this means
+                # that when the tool is called it will raise an error.
+                # so we should raise an error here.
+                if not all(getattr(_input, "tool_mode", False) for _input in inputs):
+                    non_tool_mode_inputs = [input_.name for input_ in inputs if not getattr(input_, "tool_mode", False)]
+                    msg = (
+                        f"Output '{output.name}' requires inputs that are not in tool mode. "
+                        f"The following inputs are not in tool mode: {', '.join(non_tool_mode_inputs)}. "
+                        "Please ensure all required inputs are set to tool mode."
+                    )
+                    raise ValueError(msg)
                 args_schema = create_input_schema(inputs)
             else:
-                args_schema = create_input_schema(self.component.inputs)
+                args_schema = create_input_schema(tool_mode_inputs)
             name = f"{self.component.name}.{output.method}"
             formatted_name = _format_tool_name(name)
-            tools.append(
-                StructuredTool(
-                    name=formatted_name,
-                    description=build_description(self.component, output),
-                    func=_build_output_function(self.component, output_method),
-                    args_schema=args_schema,
+            event_manager = self.component._event_manager
+            if asyncio.iscoroutinefunction(output_method):
+                tools.append(
+                    StructuredTool(
+                        name=formatted_name,
+                        description=build_description(self.component, output),
+                        coroutine=_build_output_async_function(self.component, output_method, event_manager),
+                        args_schema=args_schema,
+                        handle_tool_error=True,
+                        callbacks=callbacks,
+                    )
+                )
+            else:
+                tools.append(
+                    StructuredTool(
+                        name=formatted_name,
+                        description=build_description(self.component, output),
+                        func=_build_output_function(self.component, output_method, event_manager),
+                        args_schema=args_schema,
+                        handle_tool_error=True,
+                        callbacks=callbacks,
+                    )
                 )
+        if len(tools) == 1:
```

Comment:
This Causes Errors when only one tool for component:
Tested Edit:
This change would remove the error in components as tools for custom component.

`if len(tools) == 1 and (tool_name and tool_description)`